### ISSUE TITLE
opcda-read item limit?

### ISSUE THREAD

**Issue by saalty303** [emrebekar/node-red-contrib-opcda-client#13]
Dear all!

Is there any limitation of how many items can I read from the OPC DA server with single **opcda-read** node?
I browsed items from server during **opcda-server** configuration and found 3071 items. Then I exported these items into the json file and imported to the **opcda-read** configuration, but the communication fails in a short time and I don't know is it caused by large number of items.

greetings,

saalty

**Comment by emrebekar** [comment]
Practically there is no. But the node is using sync read. And you have to be careful when reading a large amount of data. Because, sync read will pay attention to all of the items to be read correctly. So this could be increased the read time than you wish.

**Comment by saalty303** [comment]
> Practically there is no. But the node is using sync read. And you have to be careful when reading a large amount of data. Because, sync read will pay attention to all of the items to be read correctly. So this could be increased the read time than you wish.

Thank you for your reply! I tried to read out directly the data but there is a checkbox **Cache Read**. Can this help anything if I check it? 
Do you have any recommendation about the cycle time in this amount of data?

**Comment by emrebekar** [comment]
Cache read selectable just for the read from the device or OPC DA server. So, yes it makes it faster to read but, this kind of large data from a single node could delay. 

The solution is to create multiple nodes and group your items for 100. Then, you can use the "join" node to merge payloads into a single one.

**Comment by saalty303** [comment]
> Cache read selectable just for the read from the device or OPC DA server. So, yes it makes it faster to read but, this kind of large data from a single node could delay.
> 
> The solution is to create multiple nodes and group your items for 100. Then, you can use the "join" node to merge payloads into a single one.

Yes, that was my idea to try. I mean use multiple nodes, just I'm not sure can I use same inject to read all noides in same time or need to solve to read them one-by-one for example with **cycle** node. 

Yesterday I tried to read from cache but timestamp was never changed, I don't know could it be the reason that the machine was stopped which data is read by the OPC DA server, so data were maybe unchanged on the machine as well.. I need to test it again.....

**Comment by emrebekar** [comment]
If the data doesn't change, the timestamp value will not.

**Comment by saalty303** [comment]
Yes, I suspected that. Do you know maybe do I need to force in different time the opcda-read nodes if I use more of them (inject one-by-one) or may I use the same inject node for synchronous reading?

**Comment by saalty303** [comment]
> Yes, I suspected that. Do you know maybe do I need to force in different time the opcda-read nodes if I use more of them (inject one-by-one) or may I use the same inject node for synchronous reading?

I mean do I need to take care about syncronizing like this:

![image](https://user-images.githubusercontent.com/94721231/145084094-789978a5-837c-42ac-a176-b7f91a659357.png)

Or I do not need to care about like this:

![image](https://user-images.githubusercontent.com/94721231/145084330-39bacf9b-96c7-4734-be59-8c94218e42e4.png)
